includes/reader-revenue/class-stripe-sync.php

```diff
@@ -0,0 +1,213 @@
+				$found_order = \WC_Stripe_Helper::get_order_by_charge_id( $charge->id );
+				if ( $found_order ) {
+					$order_customer_id = $found_order->get_customer_id();
+					if ( ! $order_customer_id ) {
+						// The order and the customer exist, but the order is not linked to the customer. Link them.
+						if ( ! $is_dry_run ) {
+							$found_order->set_customer_id( $user_id );
+							$found_order->save();
+							// translators: Order ID.
+							\WP_CLI::success( sprintf( __( 'Updated WC order: %d.', 'newspack' ), $found_order->get_id() ) );
+							self::$results['orders_updated'] ++;
+						}
+					}
+				} else {
+					// This is a charge without an order. Create a new order.
+					if ( ! $is_dry_run ) {
+						$wc_transaction_payload            = Stripe_Connection::create_wc_transaction_payload( $customer, $charge );
+						$wc_transaction_payload['user_id'] = $user_id;
+						$order_id                          = WooCommerce_Connection::create_transaction( $wc_transaction_payload );
+						// translators: Order ID.
+						\WP_CLI::success( sprintf( __( 'Created WC order: %d.', 'newspack' ), $order_id ) );
+						self::$results['orders_created'] ++;
+					}
+				}
+			}
+		}
+
+		self::$results['processed']++;
+	}
+
+	/**
+	 * Fetch Stripe customers.
+	 *
+	 * @param array  $args Arguments.
+	 * @param string $last_id Customer ID.
+	 */
+	private static function process_all_stripe_customers( $args, $last_id = false ) {
+		$stripe = Stripe_Connection::get_stripe_client();
+		try {
+			$params = [ 'limit' => $args['batch-size'] ];
+			if ( $last_id ) {
+				$params['starting_after'] = $last_id;
+			}
+			$response = $stripe->customers->all( $params );
+
+			// translators: Number of customers processed.
+			\WP_CLI::log( sprintf( __( 'Processing a batch of %d Stripe customers.', 'newspack' ), count( $response['data'] ) ) );
+			foreach ( $response['data'] as $customer ) {
+				self::process_stripe_customer( $customer, $args );
+			}
+
+			if ( $response['has_more'] ) {
+				$last_id             = $response['data'][ count( $response['data'] ) - 1 ]->id;
+				$intermediate_result = self::process_all_stripe_customers( $args, $last_id );
+				if ( \is_wp_error( $intermediate_result ) ) {
+					\WP_CLI::error( $intermediate_result->get_error_message() );
+				}
+			}
+		} catch ( \Throwable $e ) {
+			return new \WP_Error( 'stripe_newspack', __( 'Could not process all customers:', 'newspack' ) . ' ' . $e->getMessage() );
+		}
+	}
+
+	/**
+	 * Add CLI commands.
+	 */
+	public static function wp_cli() {
+		if ( ! defined( 'WP_CLI' ) ) {
+			return;
+		}
+
+		if ( ! class_exists( 'WC_Stripe_Helper' ) || ! method_exists( 'WC_Stripe_Helper', 'get_order_by_charge_id' ) ) {
+			\WP_CLI::error( __( 'WC Stripe Gateway plugin has to be active.', 'newspack' ) );
+			return;
+		}
+
+		if ( ! function_exists( 'wc_create_new_customer' ) ) {
+			\WP_CLI::error( __( 'WooCommerce plugin has to be active.', 'newspack' ) );
+			return;
+		}
```

Comment:
I'd move these checks into the `$sync_to_wc` function, otherwise this will throw an error when using any other CLI commands  if WooCommerce/Stripe Gateway isn't active. And it would be impossible to run `wp plugin activate woocommerce` to fix the problem! :)

Reply:
Good catch! Fixed in 6914406b19b9b879cdf6ca01d7e9d3a1ca140343